public/components/overview/components/dashboard_controls.tsx

```diff
@@ -13,19 +14,58 @@ import {
   EuiToolTip,
 } from '@elastic/eui';
 import { OnTimeChangeProps } from '@opensearch-project/oui/src/eui_components/date_picker/super_date_picker/super_date_picker';
-import React from 'react';
 import { useObservable } from 'react-use';
 import { FormattedMessage } from '@osd/i18n/react';
 import { coreRefs } from '../../../framework/core_refs';
-import { HOME_CONTENT_AREAS } from '../../../plugin_helpers/plugin_overview';
+import { HOME_CONTENT_AREAS, SECTIONS } from '../../../plugin_helpers/plugin_overview';
 import { redirectToDashboards } from '../../getting_started/components/utils';
 import { AddDashboardCallout } from './add_dashboard_callout';
+import { AddDataSourceCallout } from './add_datasource_callout';
 import { ObsDashboardStateManager } from './obs_dashboard_state_manager';
+import { SavedObjectsClientCommonFindArgs } from '../../../../../../src/plugins/data/common';
+import { getWorkspaceIdFromUrl } from '../../../../../../src/core/public/utils';
+import { setObservabilityDashboardsId } from './utils';
+import { getOverviewPage } from '../../../../common/utils';
+
+const getDatasourceAttributes = async () => {
+  const findOptions: SavedObjectsClientCommonFindArgs = {
+    type: 'data-source',
+    perPage: 1000,
+  };
+
+  const allDataSources = await coreRefs?.savedObjectsClient?.find(findOptions);
+  return allDataSources?.savedObjects ?? [];
+};
 
 export function DashboardControls() {
+  const [isDataSourceEmpty, setIsDataSourceEmpty] = useState<boolean | null>(null);
+  const [isInWorkspace, setIsInWorkspace] = useState<boolean>(false); // Track if user is in a workspace
   const isDashboardSelected = useObservable(ObsDashboardStateManager.isDashboardSelected$);
   const dashboardState = useObservable(ObsDashboardStateManager.dashboardState$);
 
+  useEffect(() => {
+    const checkDataSource = async () => {
```

Comment:
We should move this function definition out of the useEffect to make it clearer

Reply:
Done